ivadomed/preprocessing.py

```diff
@@ -0,0 +1,46 @@
+import copy
+
+
+def get_preprocessing_transforms(transforms):
+    original_transforms = copy.deepcopy(transforms)
+    preprocessing_transforms = copy.deepcopy(transforms)
+    for idx, tr in enumerate(original_transforms):
+        if "preprocessing" in transforms[tr] and transforms[tr]["preprocessing"]:
+            del transforms[tr]
+        else:
+            del preprocessing_transforms[tr]
+
+    return preprocessing_transforms
+
+
+def apply_transforms(transforms, seg_pair, roi_pair=None):
+    if transforms is None:
+        return (seg_pair, roi_pair)
+
+    if roi_pair is not None:
+        stack_roi, metadata_roi = transforms(sample=roi_pair["gt"],
+                                             metadata=roi_pair['gt_metadata'],
+                                             data_type="roi")
```

Comment:
We would also need to spread the params from metadata_roi (where we compute the crop params for ROICrop to metdata_input and metadata_gt --> via `update_medaata`